exonum-java-binding/core/rust/src/storage/proof_map_index.rs

```diff
@@ -63,18 +88,34 @@ pub extern "system" fn Java_com_exonum_binding_core_storage_indices_ProofMapInde
     _: JClass,
     name: JString,
     view_handle: Handle,
-    // TODO: to be used in ECR-3765
-    _key_hashing: jboolean,
+    key_hashing: jboolean,
 ) -> Handle {
     let res = panic::catch_unwind(|| {
         let name = utils::convert_to_string(&env, name)?;
+        let key_is_hashed = key_hashing != JNI_FALSE;
```

Comment:
Will `as bool` work?

Reply:
Perhaps, but usually it is recommended comparing to `JNI_TRUE`. I believe that `jboolean != JNI_FALSE` is even safer 🙃.

Reply:
Why is it safer? I'd better prefer `should_hash_key = should_hash_key == JNI_TRUE`, just for better readability. Parsing boolean expressions isn't my best side.

Reply:
Ok, switched to comparing against `JNI_TRUE`.